libraries/apollo-runtime/src/commonTest/kotlin/test/network/NetworkMonitorTest.kt

```diff
@@ -107,3 +111,19 @@ class NetworkMonitorInterceptor(private val networkMonitor: NetworkMonitor): Apo
   }
 }
 
+class MockServerTest(val mockServer: MockServer, val apolloClient: ApolloClient, val scope: CoroutineScope)
+
+fun mockServerTest(
+    clientBuilder: ApolloClient.Builder.() -> Unit = {},
+    block: suspend MockServerTest.() -> Unit
+) = runTest(true) {
+  MockServer().use { mockServer ->
+    ApolloClient.Builder()
+        .serverUrl(mockServer.url())
+        .apply(clientBuilder)
+        .build()
+        .use {apolloClient ->
+          MockServerTest(mockServer, apolloClient, this).block()
+        }
+  }
+}
```

Comment:
This code is duplicated a bunch of times. It's a bit sad but it's very trivial code so it shouldn't be too hard to maintain.